### DISCUSSION TITLE
Como recuperar os dados dos Secure Fields do Método Core no backend?

### DISCUSSION THREAD

**Discussion by joaofaveri** [mercadopago/sdk-react#111] [Q&A]
Estou fazendo a integração da API do Mercado Pago com minha aplicação e pretendo utilizar os Secure Fields do SDK react para coletar os dados sensíveis do cliente, gerar o cardToken e enviar para o backend, seja para salvar o id do cartão no perfil do cliente, seja para vincular esse card id a uma assinatura. Ocorre que cada Secure Fields inclui um iframe na aplicação para cada campo e não estou conseguindo recuperar os dados preenchido no formulário... 

Estou seguindo a orientação da página [Métodos Core](https://www.mercadopago.com.br/developers/pt/docs/sdks-library/client-side/sdk-js-react/core-methods-installation), mas nada dá certo...

**Comment by francineemilia** [comment]
Olá @joaofaveri

[Secure Fields](https://github.com/mercadopago/sdk-react?tab=readme-ov-file#secure-fields) é uma solução diferente de [Métodos Core](https://github.com/mercadopago/sdk-react?tab=readme-ov-file#core-methods). Secure Fields de fato inclui um iframe, pois ele visa isolar os elementos do cartão para garantir a segurança destes dados durante a transação. Neste caso, vale verificar o método de [Criar card token](https://github.com/mercadopago/sdk-react?tab=readme-ov-file#createcardtoken) com secure fields.

**Reply by joaofaveri** [reply]
Agradeço a gentileza da resposta. Ficou claro a distinção entre os Secure Fields e Core Methods. Agora, Para ver se entendi, utilizando o createCardToken dos Secure Fields, eu só preciso passar as informações do cardholder, Identification Type e Number? Os demais dados, que estão nos Secure Fields (card Number, expiration date e security code) o método busca automaticamente?

**Reply by francineemilia** [reply]
@joaofaveri do que entendo os demais dados já ficam "visíveis" ao método de criação do card token quando são preenchidos em secure fields. Quando observamos os campos solicitados, são justamente os que não são PCI para uma transação: cardholderName, identificationType e identificationNumber.

Ressalto que pode ser interessante utilizar uma solução como o [Payment Brick](https://github.com/mercadopago/sdk-react?tab=readme-ov-file#payment-brick) ou o [Card Payment Brick](https://github.com/mercadopago/sdk-react?tab=readme-ov-file#card-payment-brick). Isso porque ele já utiliza o Secure Fields e tem os campos necessários de dados não PCI ("não sensíveis") para gerar o cardToken, enviando o mesmo com outros dados para seu backend na função de submit.

**Reply by joaofaveri** [reply]
Perfeito! Fiz os testes aqui e realmente o método de criação do token "pega" os dados dos secure fields, necessitando apenas complementar com os campos não PCI. Estou utilizando essa alternativa porque estou apenas salvando o cartão do cliente para uso posterior num fluxo de assinatura, e as soluções Bricks, pelo que entendi, exigem a inicialização com um valor de pagamento, o que não é o caso aqui
